Description: Honour the SOURCE_DATE_EPOCH environment variable
 Honour the SOURCE_DATE_EPOCH environment variable, so that output documents'
 timestamp is set to last debian/changelog entry when building packages that
 use asciidoctor in their building process.
Author: Alexis Bienvenüe <pado@passoire.fr>
Forwarded: https://github.com/asciidoctor/asciidoctor/pull/1721
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=820435
Reviewed-by: Cédric Boutillier <boutil@debian.org>
Last-Update: 2016-06-08


Index: asciidoctor-1.5.4/lib/asciidoctor/document.rb
===================================================================
--- asciidoctor-1.5.4.orig/lib/asciidoctor/document.rb
+++ asciidoctor-1.5.4/lib/asciidoctor/document.rb
@@ -401,7 +401,11 @@ class Document < AbstractBlock
       #attrs['infile'] = attrs['docfile']
 
       # dynamic intrinstic attribute values
-      now = ::Time.now
+      if ENV['SOURCE_DATE_EPOCH'].nil?
+        now = ::Time.now
+      else
+        now = ::Time.at(ENV['SOURCE_DATE_EPOCH'].to_i).gmtime
+      end
       localdate = (attrs['localdate'] ||= now.strftime('%Y-%m-%d'))
       unless (localtime = attrs['localtime'])
         begin
Index: asciidoctor-1.5.4/lib/asciidoctor.rb
===================================================================
--- asciidoctor-1.5.4.orig/lib/asciidoctor.rb
+++ asciidoctor-1.5.4/lib/asciidoctor.rb
@@ -1308,7 +1308,11 @@ module Asciidoctor
     if ::File === input
       # TODO cli checks if input path can be read and is file, but might want to add check to API
       input_path = ::File.expand_path input.path
-      input_mtime = input.mtime
+      if ENV['SOURCE_DATE_EPOCH'].nil?
+        input_mtime = input.mtime
+      else
+        input_mtime = ::Time.at(ENV['SOURCE_DATE_EPOCH'].to_i).gmtime
+      end
       lines = input.readlines
       # hold off on setting infile and indir until we get a better sense of their purpose
       attributes['docfile'] = input_path
